src/control-plane/backend/lambda/api/model/pipeline.ts

```diff
@@ -1300,10 +1300,13 @@ export class CPipeline {
         plugins: plugins,
       };
     }
-    const transformPlugins = this.resources.plugins?.filter(plugin => plugin.id === this.pipeline.dataProcessing?.transformPlugin);
+    let transformPlugin = this.resources.plugins?.find(plugin => plugin.id === 'BUILT-IN-1');
+    if (this.pipeline.dataProcessing?.transformPlugin) {
+      transformPlugin = this.resources.plugins?.find(plugin => plugin.id === this.pipeline.dataProcessing?.transformPlugin);
+    }
     const enrichPlugin = this.resources.plugins?.filter(plugin => this.pipeline.dataProcessing?.enrichPlugin?.includes(plugin.id));
     return {
-      transformPlugin: transformPlugins?.length === 1? transformPlugins[0] : null,
+      transformPlugin,
```

Comment:
Do we have test case to verify this output change?

Reply:
Yes, there had a case covering this change.